components/lib/parseMaterial.js

```diff
@@ -0,0 +1,34 @@
+import { processColor } from 'react-native';
+import { intersection } from 'lodash';
+
+// https://developer.apple.com/documentation/scenekit/scnmaterial
+const materialPropertiesWithMaps = [
+  'normal',
+  'diffuse',
+  'displacement',
+  'specular',
+];
+
+export function processMaterialPropertyContents(material) {
+  const propsToUpdate = intersection(
+    Object.keys(material),
+    materialPropertiesWithMaps,
+  );
+
+  return propsToUpdate.reduce(
+    (prev, curr) => ({
+      ...prev,
+      [curr]: {
+        ...prev[curr],
+        color: processColor(
+          curr === 'diffuse' && typeof prev[curr] === 'string'
+            ? prev[curr]
+            : prev[curr].color,
+        ),
+      },
+    }),
+    material,
+  );
```

Comment:
- it took my a while to get that. Can you rename the variables? Also maybe an approach without reduce would be more easy to understand. maybe with mapValues from lodash?
- also a comment about why we support diffuse to be a string.
- instead of typeof you can use `isString(prevMaterial[property])`